restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/IncomingPhoneNumbersEndpoint.java

```diff
@@ -261,14 +268,62 @@ protected Response getAvailableCountries(final String accountSid, final MediaTyp
         }
     }
 
-    protected Response getIncomingPhoneNumbers(final String accountSid, final String phoneNumberFilter, final String friendlyNameFilter,
-            PhoneNumberType phoneNumberType, final MediaType responseType) {
+    protected Response getIncomingPhoneNumbers(final String accountSid, final PhoneNumberType phoneNumberType, UriInfo info,
+            final MediaType responseType) {
+
         secure(accountsDao.getAccount(accountSid), "RestComm:Read:IncomingPhoneNumbers");
-        IncomingPhoneNumberFilter incomingPhoneNumberFilter = new IncomingPhoneNumberFilter(accountSid, friendlyNameFilter, phoneNumberFilter);
+
+        String phoneNumberFilter = info.getQueryParameters().getFirst("PhoneNumber");
+        String friendlyNameFilter = info.getQueryParameters().getFirst("FriendlyName");
+        String page = info.getQueryParameters().getFirst("Page");
+        String reverse = info.getQueryParameters().getFirst("Reverse");
+        String pageSize = info.getQueryParameters().getFirst("PageSize");
+
+        if (pageSize == null) {
+            pageSize = "50";
+        }
+
+        if (page == null) {
+            page = "0";
+        }
+
+        int limit = Integer.parseInt(pageSize);
+        int offset = (page == "0") ? 0
+                : (((Integer.parseInt(page) - 1) * Integer.parseInt(pageSize)) + Integer.parseInt(pageSize));
```

Comment:
can we reduce the number of times we convert incoming queryString params into proper type for validation?? For performance and error handling point of view good be much better...